### ISSUE TITLE
Enable 'Bios Selection' Option, Before Run A Game.

### ISSUE THREAD

**Issue by rslkr** [yoshisuga/MAME4iOS#397]
Hi, Todd, I have another idea (I'll close this if you don't like it)

Latest MAME has the Bios Selection option in the MAME config menu.
If a game requires Bios, it appears on the list, same goes for the Dipswitch...

In my opinion, MAME config Menu well designed,
Dipswitch and Bios require a 'Reset' to activate the settings. They (MAMEDEV) added the RESET option, in their list (Dipswitch and Bios)
As far as I know, both 'Soft Reset and Hard Reset' work for Dipswitch, but only 'Hard Reset' works for Bios.

For the Dipswitch, once you adjust it, MAME saves the settings in a game cfg (I think), so no need to do it again.

Long time ago, I used to do this Bios settings in 'mame.ini' or only for a specific game, used to edit (game name).ini file.

Now, if I launch the MAME with command line, MAME shows a Bios option before run a game.

My request is, can you add this option on the 'Game Choose UX' Game's Title long press Popup Menu?

Its an old thread but, hope this helps: https://dammit.typepad.com/blog/2011/03/setting-the-neogeo-bios-in-mame.html

Like you mentioned before, **Dipswitch is so complicated**, MAME reads game's Dipswitch options from the games drivers and also some games Dipswitch have 'F2' Service in it.

I was thinking something like this: '**Adjust Game**' section on the Popup Menu for Dipswitch and Bios
But, it seems NOT possible for Dipswitch menu, 'coz a game has to run first!!!

**Comment by ToddLa** [comment]
Yes the inspirational goal is to avoid the need to use the `MAME` "blue menu", see issue #186

Yes the right place for UX you describe is in the context menu for the game/romset.

Other things to have UX for, in context menu...
* Metadata like Title, Description, Image, etc.
* Controller/Key mappings. 
* Dipswitch and other machine options. 
  - device options, joysticks attached
  - media or disk image attachments. 
* Ability to edit Per Game, per Romset, per Machine, or per Perent, or Global default.
